torchquad/integration/newton_cotes.py

```diff
@@ -47,8 +49,13 @@ def calculate_result(self, function_values, dim, n_per_dim, hs):
             backend tensor: Quadrature result
         """
         # Reshape the output to be [N,N,...] points instead of [dim*N] points
-        function_values = function_values.reshape([n_per_dim] * dim)
-
+        integrand_shape = function_values.shape[1:]
+        dim_shape = [n_per_dim] * dim
+        new_shape = [*integrand_shape, *dim_shape]
+        einsum = "".join([chr(i + 65) for i in range(len(function_values.shape))])
```

Comment:
could you add a comment explaining what this line does? :)

Reply:
(and maybe the two below)

Reply:
done!